### ISSUE TITLE
Please add SVG support

### ISSUE THREAD

**Issue by Emasoft** [CreateJS/EaselJS#456]
I use SVG images in many projects because it is resolution independent and portable on different devices and screens dpi. Also the artists I work with can save natively in SVG format from graphical applications like Inkscape, Adobe Illustrator and Edge Animate. EaselJS is a great library for animations but I cannot find any support for SVG graphics. There is a way to convert the SVG to a raster image using the Bitmap class:
http://createjs.com/Docs/EaselJS/classes/Bitmap.html 
But this defies and frustrate the use of SVG, losing the device independency on retina displays and all the interactivity and the animations of the SVG images.
Can you add a native SVG support in EaselJS without the need to rasterize SVG? Thanks.

**Comment by TheAifam5** [comment]
Any updates and infos?

**Comment by scoop2** [comment]
this guy was able to port SVG animation to the canvas using EaselJS
http://jasonwritescode.blogspot.de/2014/05/moving-my-svg-animation-to-canvas-with.html

he pointed to a parser:
http://www.professorcloud.com/svg-to-canvas/

**Comment by trusktr** [comment]
What's the status of this? Can we draw SVG with EaselJS now?

**Comment by janisdd** [comment]
Maybe not all svg features are supported but the docs https://www.createjs.com/docs/easeljs/classes/Bitmap.html claim that you can use svg.
And for me it seems working, example https://jsfiddle.net/janisdd/cuqgztLr/20/

**Comment by reyco1** [comment]
I completely agree! It would be great though if there were an
implementation of SVG->createjs.Bitmap similar to how there is an
implementation of video->createjs.Bitmap in the sense that we would be able
display an animated SVG.

I am currently achieving this by using BodyMovin. I export an animation
from AfterEffects as JSON data that then gets consumed by a BodyMovin
instance that renders that animation into a canvas. I then feed that canvas
element into a createjs.Bitmap instance and display it on the stage. It
works great, I just wish there were a more straight forward solution within
CreateJS itself.

**Comment by reyco1** [comment]
I completely agree! It would be great though if there were an implementation of SVG->createjs.Bitmap similar to how there is an implementation of video->createjs.Bitmap in the sense that we would be able display an animated SVG.

I am currently achieving this by using BodyMovin. I export an animation from AfterEffects as JSON data that then gets consumed by a BodyMovin instance that renders that animation into a canvas. I then feed that canvas element into a createjs.Bitmap instance and display it on the stage. It works great, I just wish there were a more straight forward solution within CreateJS itself.

**Comment by EtagiBI** [comment]
> Maybe not all svg features are supported but the docs https://www.createjs.com/docs/easeljs/classes/Bitmap.html claim that you can use svg.
> And for me it seems working, example https://jsfiddle.net/janisdd/cuqgztLr/33/
It imports SVG as a raster image. Try to scale down / up your SVG to see clipping / pixelization.

**Comment by danzen** [comment]
If it is of any help to folks - we converted SVG to Squiggle and Blob objects in ZIM - powered by CreateJS.  We put the shapes into a Container called an SVGContainer that extends a ZIM Container that extends a CreateJS Container.  You can see the docs here https://zimjs.com/docs.html?item=SVGContainer.  At the bottom the docs for the entry is a VIEW link that will show the conversion code - it is a little messy as we had two of us work on it and we are considering it experimental as SVG is quite a large world of options.  Here is a demonstration http://zimjs.com/svg/index.html and then our test page https://zimjs.com/explore/svgcontainer2.html - this becomes pretty powerful as they can be user controlled curves, we can animate along them, drag things along them, add Text along them, do shape tweens and hitTestPath (new in ZIM 10.3.0).
